test/groovy/com/sap/piper/MtaUtilsTest.groovy

```diff
@@ -0,0 +1,247 @@
+package com.sap.piper
+
+import org.junit.Before
+import org.junit.ClassRule
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.ExpectedException
+import org.junit.rules.RuleChain
+import org.junit.rules.TemporaryFolder
+import org.yaml.snakeyaml.Yaml
+
+import com.lesfurets.jenkins.unit.BasePipelineTest
+
+import groovy.json.JsonSlurper
+import hudson.AbortException
+import util.JenkinsEnvironmentRule
+import util.Rules
+
+
+class MtaUtilsTest extends BasePipelineTest {
+    private static srcPackageJson = 'test/resources/MtaUtils/package.json'
+    private static mtaTemplate = 'resources/template_mta.yml'
+    private static data
+    private static targetMtaDescriptor
+    private File badJson
+    private mtaUtils
+
+    @Rule
+    public JenkinsEnvironmentRule jer = new JenkinsEnvironmentRule(this)
+
+
+    @Rule
+    public ExpectedException thrown= ExpectedException.none();
+
+    @ClassRule
+    public static TemporaryFolder tmp = new TemporaryFolder()
+
+    @Rule
+    public RuleChain ruleChain = Rules
+    .getCommonRules(this)
+    .around(jer)
+
+    void prepareObjectInterceptors(object) {
+        object.metaClass.invokeMethod = helper.getMethodInterceptor()
+        object.metaClass.static.invokeMethod = helper.getMethodInterceptor()
+        object.metaClass.methodMissing = helper.getMethodMissingInterceptor()
+    }
+
+    @Before
+    void init() {
+        targetMtaDescriptor = "${tmp.getRoot()}/generated_mta.yml"
+        def script = new Object()
+        mtaUtils = new MtaUtils(script)
+
+        prepareObjectInterceptors(script)
+
+        this.helper.registerAllowedMethod('readJSON', [Map], { Map parameters ->
+            def js = new JsonSlurper()
+            def reader = new BufferedReader(new FileReader(parameters.file))
```

Comment:
js.parse is also able to deal with files directly, e.g. ```js.parse(new File(parameters.file))```. Using that approach it is not required to deal with the readers. And in case it is preferred to use the readers anyway: The ```reader``` is not closed. Unlikely that this gets a real issue here from the test, but at least it causes FindBugs warnings.

Reply:
reworked.
Btw, the code snippet exists also in the MockHelper.groovy